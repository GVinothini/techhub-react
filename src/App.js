import logo from "./logo.svg";
import "./App.css";
import { Component } from "react";
import UserTable from "./UserTable";
import { connect } from "react-redux";

class App extends Component {
    constructor(props) {
        super(props);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleSubmit(event) {
        event.preventDefault();
        const formElement = event.target;
        const formData = new FormData(formElement);
        const userName = formData.get("InputName");
        this.props.addNewUsername(userName);
    }

    render() {
        let buttonClasses = `btn ${this.props.buttonTheme}`;
        return (
            <div className="container">
                <div className="row">
                    <div className="col-xs-6 col-xs-offset-3">
                        <div className="row">
                            <div className="col-xs-12">
                                <UserTable userList={this.props.userList} />
                            </div>
                        </div>
                        <div className="row">
                            <div className="col-xs-12">
                                <form
                                    className="form-inline"
                                    onSubmit={this.handleSubmit}
                                >
                                    <div className="form-group">
                                        <label htmlFor="InputName1">Name</label>
                                        <input
                                            type="text"
                                            className="form-control"
                                            id="InputName1"
                                            name="InputName"
                                            placeholder="Jane Doe"
                                        />
                                    </div>
                                    <button
                                        type="submit"
                                        className={buttonClasses}
                                    >
                                        Save
                                    </button>
                                </form>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}

const mapStateToProps = (state, ownProps) => {
    return {
        userList: state.userList,
        buttonTheme: ownProps.buttonTheme,
    };
};

const mapDispatchToProps = (dispatch) => ({
    addNewUsername: (userName) =>
        dispatch({
            type: "ADD_USER",
            payload: userName,
        }),
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
