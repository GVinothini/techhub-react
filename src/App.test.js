import React from 'react';
import '@testing-library/jest-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import App from "./App";

jest.mock('react-redux', () => ({
    connect: (mapStateToProps, mapDispatchToProps) => (reactComponent) => ({
        mapStateToProps,
        mapDispatchToProps: (dispatch = jest.fn((action) => action), ownProps) => mapDispatchToProps(dispatch, ownProps),
        reactComponent,
        mockDispatch: jest.fn((action) => action)
    })
}));

jest.mock('./UserTableWithDelete', () => () => <div>Table</div>);

describe('App', () => {
    const mockProps = {
        userList: [ 'test-user1', 'test-user2'],
        addNewUsername: jest.fn()
    };
    it('should render app', () => {
       const { container } = render(<App.reactComponent {...mockProps} />);
       expect(container).toMatchSnapshot();
    });

    it('should call addNewUsername while click save button', () => {
        render(<App.reactComponent {...mockProps} />);
        fireEvent.click(screen.getByRole('button', { name: 'Save' }));
        expect(mockProps.addNewUsername).toBeCalled();
    });

    it('should call mapStateToProps', () => {
        const mockState = { userList: [ 'test-user1', 'test-user2'] };
        const ownProps = { buttonTheme: 'button-theme'};
        const result = App.mapStateToProps(mockState, ownProps);
        expect(result).toEqual({
            userList: mockState.userList,
            buttonTheme: ownProps.buttonTheme
        });
    });

    it('should call mapDispatchToProps', () => {
        const dispatch = jest.fn();
        const result = App.mapDispatchToProps(dispatch);
        result.addNewUsername();
        result.deleteUser();
    });
});
