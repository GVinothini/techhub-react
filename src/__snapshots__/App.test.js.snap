// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render app 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="row"
    >
      <div
        class="col-xs-6 col-xs-offset-3"
      >
        <div
          class="row"
        >
          <div
            class="col-xs-12"
          >
            <div>
              Table
            </div>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col-xs-12"
          >
            <form
              class="form-inline"
            >
              <div
                class="form-group"
              >
                <label
                  for="InputName1"
                >
                  Name
                </label>
                <input
                  class="form-control"
                  id="InputName1"
                  name="InputName"
                  placeholder="Jane Doe"
                  type="text"
                />
              </div>
              <button
                class="btn undefined"
                type="submit"
              >
                Save
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
